util/network-devp2p/src/session.rs

```diff
@@ -104,29 +104,45 @@ impl Session {
 		nonce: &H256, host: &HostInfo) -> Result<Session, Error>
 		where Message: Send + Clone + Sync + 'static {
 		let originated = id.is_some();
-		let mut handshake = Handshake::new(token, id, socket, nonce).expect("Can't create handshake");
-		let local_addr = handshake.connection.local_addr_str();
-		handshake.start(io, host, originated)?;
-		Ok(Session {
-			state: State::Handshake(handshake),
-			had_hello: false,
-			info: SessionInfo {
-				id: id.cloned(),
-				client_version: String::new(),
-				protocol_version: 0,
-				capabilities: Vec::new(),
-				peer_capabilities: Vec::new(),
-				ping: None,
-				originated: originated,
-				remote_address: "Handshake".to_owned(),
-				local_address: local_addr,
+		match Handshake::new(token, id, socket, nonce) {
+			Ok(mut handshake) => {
+				let local_addr = handshake.connection.local_addr_str();
+				handshake.start(io, host, originated)?;
+				Ok(Session {
+					state: State::Handshake(handshake),
+					had_hello: false,
+					info: SessionInfo {
+						id: id.cloned(),
+						client_version: String::new(),
+						protocol_version: 0,
+						capabilities: Vec::new(),
+						peer_capabilities: Vec::new(),
+						ping: None,
+						originated: originated,
+						remote_address: "Handshake".to_owned(),
+						local_address: local_addr,
+					},
+					ping_time: Instant::now(),
+					pong_time: None,
+					expired: false,
+					protocol_states: HashMap::new(),
+					compression: false,
+				})
 			},
-			ping_time: Instant::now(),
-			pong_time: None,
-			expired: false,
-			protocol_states: HashMap::new(),
-			compression: false,
-		})
+			Err(err) => {
+				if let Error(ErrorKind::Io(inner), state) = err {
+					Err(Error(
+						if inner.raw_os_error() == Some(24) {
+							// handle sys error CODE 24; "Too many open files" more gracefully
+							ErrorKind::TooManyOpenFiles("Can't create handshake.".to_string())
```

Comment:
Forgive me the silly question but what does this look like to the user in the console?

```
TooManyOpenFiles: Can't create handshake
```

Does the process exit at that point? Does it carry with it the error code when exiting? I'm thinking about people running parity on a server with monitoring attached looking for specific process exit codes to trigger alarms.

Reply:
Honestly wasn't able to trigger this particular warning myself, but took the information provided in the corresponding ticket - https://github.com/paritytech/parity/issues/6791 .

From my understanding this happens in a Worker-Thread, which - as a result - then panics. The process itself keeps running and from the weirdness of the backtrace message, I doubt anyone tried to have any script do things when that error appears. So for the case you are thinking about this doesn't change anything.